File: Anul II/Semestrul II/Medii de proiectare si programare/AgentiiZboruri/Persistence/src/main/java/ro/persistence/DbUtils.java
package ro.persistence;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;
import java.util.Properties;

public class DbUtils extends Properties {
    private final Properties jdbcProps;

    public DbUtils(Properties props) {
        this.jdbcProps = props;
    }

    private Connection instance = null;

    private Connection getNewConnection() {
        String url = jdbcProps.getProperty("jdbc.url");
        String user = jdbcProps.getProperty("jdbc.user");
        String pass = jdbcProps.getProperty("jdbc.password");
        Connection con = null;
        try {
            if (user != null && pass != null)
                con = DriverManager.getConnection(url, user, pass);
            else
                con = DriverManager.getConnection(url);
        } catch (SQLException ex) {
            System.out.println("Error getting connection to the database " + ex);
        }
        return con;
    }

    public Connection getConnection() {
        try {
            if (instance == null || instance.isClosed())
                instance = getNewConnection();
        } catch (SQLException ex) {
            System.out.println("Error DB " + ex);
        }
        return instance;
    }
}

File: Anul II/Semestrul II/Medii de proiectare si programare/ClientServer/Persistence/DbUtils.cs
using System.Collections.Specialized;
using System.Data.SqlClient;

namespace Persistence
{
    public class DbUtils
    {
        private static readonly Logger Logger = LogManager.GetCurrentClassLogger();
        private readonly NameValueCollection _jdbcProps;
        private SqlConnection _instance;

        public DbUtils(NameValueCollection props)
        {
            _jdbcProps = props;
        }

        private SqlConnection GetNewConnection()
        {
            Logger.Trace("Entering GetNewConnection");
            var url = _jdbcProps["jdbc.url"];
            var user = _jdbcProps["jdbc.user"];
            var pass = _jdbcProps["jdbc.password"];
            SqlConnection con = null;
            try
            {
                con = string.IsNullOrEmpty(user) || string.IsNullOrEmpty(pass) 
                    ? new SqlConnection(url) 
                    : new SqlConnection(url, new SqlCredential(user, pass));
                con.Open();
            }
            catch (SqlException ex)
            {
                Logger.Error(ex, "Error getting connection to the database");
                System.Console.WriteLine($"Error getting connection to the database {ex}");
            }
            Logger.Trace("Exiting GetNewConnection");
            return con;
        }

        public SqlConnection GetConnection()
        {
            Logger.Trace("Entering GetConnection");
            try
            {
                if (_instance == null || _instance.State == System.Data.ConnectionState.Closed)
                    _instance = GetNewConnection();
            }
            catch (SqlException ex)
            {
                Logger.Error(ex, "Error DB");
                System.Console.WriteLine($"Error DB {ex}");
            }
            Logger.Trace("Exiting GetConnection");
            return _instance;
        }
    }
}